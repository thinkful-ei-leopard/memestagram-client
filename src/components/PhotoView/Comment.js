import React, { Component } from 'react'
import './PhotoView.css'


export default class Comment extends Component {
    render() {
        return (
            <div>
                <span>{this.props.comments.username}</span>
                <p>{this.props.comments.comment}</p>
            </div>
        )
    }
}
