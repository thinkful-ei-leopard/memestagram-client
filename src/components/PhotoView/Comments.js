import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class Comments extends Component {
    render() {
        return (
            <div className='comments' >             
                    <p className='comment'><Link to={`/users/${this.props.user_id}`}> <span className='user'>{this.props.user}</span> </Link> {this.props.comment}</p>        
            </div>
        )
    }
}
