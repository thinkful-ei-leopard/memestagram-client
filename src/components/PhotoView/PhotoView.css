#newComment{
    width: 100%;
    margin-left: 0px;
    height: 40px;
    padding-right: 70px;
    padding-left: 5px;
}
#post{
    margin-left: -70px;
    height: 20px;
    color:#042f4b;
    padding: 0;
    background-color: white;
    border: none;
    margin-right: 0;
    width: 50px;
    border-radius: 0%;
}
#post:hover {  
    color:#ed1250;   
}

.memePost{
    max-width: 500px;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 20px;
    background-color: white;
    color:#042f4b;
}
.user_name{
    display: inline-block;
    font-weight: normal;
    margin-right: 15px;
    color:#042f4b;     
}

.namedes{
  text-align: left;
}

.first{
    display: flex;
    justify-content: space-between;
}
.first > .delete{
    margin: 10px;
}
.before:hover{
    color:#ed1250 ;
}
.des{
    font-size: 16px;
    margin-left: 10px; 
}
.photoview{
    margin-left: 10px;
    float: right;
}
.delete:hover{
    color:#ed1250 ;
}

.before{
  margin-right: 5px;
}
.user{
    margin-right: 10px;
}

.heart{
    display: inline-block;
}

.comments{
  display: flex;
  margin-left: 10px;
  margin-bottom: 0px;
}
.comment > a{
    color:#042f4b;
}
.comment{
    margin-top: 0;
    
}
.comment > a:hover{
    color:#ed1250;
}

.postComment{
    border: none;
    padding: 18px;
}
.commentdiv{
    margin-right: 6px;
}
.total{
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-weight: bold;
}

.content{
   padding:10px ;
}
@media screen and (min-width: 481px){
    .photoview{
        margin: 10px;
        float: right;
        font-size: 16px;
    }
    
}

@media screen and (max-width: 365px){
  .content >.delete{
      margin-top:15px ;
  }
}
