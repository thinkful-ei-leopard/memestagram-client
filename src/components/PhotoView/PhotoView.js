import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import NavBar from '../NavBar/NavBar';
import CommentsService from '../../services/comment-service';
import PostsService from '../../services/posts-service';
import MemeContext from '../../context/MemeContext';
import { Input, Label } from '../Form/Form';
import Button from '../Button/Button';
import Comments from './Comments';

import './PhotoView.css';
export default class PhotoView extends Component {
    
    static defaultProps={
        history:{
            push:()=>{}
        },
        handleUserView: () => { }
    }  

    state = {
        postId: 0,
        error: null,
        singlePost: '',
        comments: [],
        newComment: null
    }

    static contextType = MemeContext

    componentDidMount(){
    const {postId }=this.props.match.params
    this.setState({postId: Number(postId)})
        this.context.clearError()
        PostsService.getOnePost(postId)
            .then(res => this.setState({singlePost:res}))
            .catch(this.context.setError)
        CommentsService.getComment(postId)
            .then(res => this.setState({comments:res}))
            .catch(this.context.setError)        
    }

    handleUsernameClick = () => {
        this.props.handleUserView()
    }

    async addComment(e){
        e.preventDefault();  
        const {newComment}=e.target

        await CommentsService.postComment(
        newComment.value,
        this.state.postId, this.context.user.id)
        .then(data =>{
            newComment.value=' '
        })
        .catch(res =>{
            this.setState({
                error:JSON.stringify(res.error)
            })
        })

        await  CommentsService.getComment(this.state.postId)
        .then(res => this.setState({comments:res}))
        .catch(this.context.setError) 
    }

    async handleDelete(postId, e) {
        e.preventDefault()
        await PostsService.deletePost(postId)
            .then(this.setState({delete: true}))
        await this.props.history.push('/dashboard');
    }

    renderComments(){
        const comments= this.state.comments;
        const commentsArr=[];
            for(let i = 0; i < comments.length; i++) {
                commentsArr.push(
                    <Comments key={comments[i].id} user={comments[i].username} user_id={comments[i].user_id} comment={comments[i].comment}/>
                )
            }
        

        return commentsArr
    }

    renderDeletePost(post) {
        if(post.user_id === this.context.user.id) {
            return <div  className='delete'> <span role='img' aria-label='delete' className='delete-emoji' onClick={e => { if (window.confirm('Are you sure you want to delete this meme?')) this.handleDelete(this.state.postId, e)}}>üóëÔ∏è</span> </div>
        } else {
            return
        }
    }

    render() {
        const {singlePost, comments}=this.state
        
        return (
            <div>
                <NavBar />
                <div className='memePost'>
                    <img src={singlePost.memeImg} alt='meme' className='meme' />
                </div>
                <div className='image-cropper single'>
                <img src={singlePost.userImg} alt='meme' className='user-img' />
                </div>
                <Link to={`/users/${this.state.singlePost.user_id}`} onClick={this.handleUsernameClick}> <h3 className='user_name'>{singlePost.username}</h3> </Link>
                <p className='user_name'>{singlePost.description}</p> 
                <p>Total Comments: {comments.length}</p>
                <div>{this.renderDeletePost(singlePost)}</div>
                <div>{this.renderComments()}</div> 
                <form onSubmit={(e)=>this.addComment(e)}>
                <div className='postComment'>
                <Label htmlFor='newComment'></Label>
                <Input type='text' name='newComment' id='newComment' placeholder=' Add comment ...' required></Input>
                <Button type='submit' id='post' >Post</Button>
                </div> 
                </form>           
            </div>
        )
    }
}
