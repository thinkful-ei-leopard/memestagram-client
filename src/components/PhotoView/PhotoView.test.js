import React from 'react'
import ReactDom from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PhotoView from './PhotoView'
import Comments from './Comments'
import { BrowserRouter } from 'react-router-dom'

describe('PhotoView Page Component', () => {
  
  const defaultProps = {
    match: {params: {postId: 1}}
  }


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <BrowserRouter>
        <PhotoView 
          {...defaultProps}
        />
      </BrowserRouter>,
      div
    );
    ReactDom.unmountComponentAtNode(div);
  });

  it('renders PhotoView', () => {
    const comments = {
      id: 1,
      user_id: 1,
      username: 'test',
      comment: 'testststs'
    }

    const wrapper = shallow(
      <BrowserRouter>
        <PhotoView {...defaultProps}>
          <Comments 
            key={comments.id} user={comments.username} user_id={comments.user_id} comment={comments.comment}
          />
        </PhotoView>
      </BrowserRouter>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})