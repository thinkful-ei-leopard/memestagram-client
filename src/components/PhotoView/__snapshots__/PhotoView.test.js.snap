// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoView Page Component renders PhotoView 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PhotoView
    handleUserView={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "postId": 1,
        },
      }
    }
  >
    <Comments
      comment="testststs"
      key="1"
      user="test"
      user_id={1}
    />
  </PhotoView>
</Router>
`;
